charts/newrelic-infrastructure-v3/templates/_helpers.tpl

```diff
@@ -153,6 +153,31 @@ Returns fargate
 {{- end -}}
 {{- end -}}
 
+{{/*
+Returns lowDataMode
+*/}}
+{{- define "newrelic.lowDataMode" -}}
+{{/* `get` will return "" (empty string) if value is not found, and the value otherwise, so we can type-assert with kindIs */}}
+{{- if (get .Values "lowDataMode" | kindIs "bool") -}}
+  {{- if .Values.lowDataMode -}}
+    {{/*
+        We want only to return when this is true, returning `false` here will template "false" (string) when doing
+        an `(include "newrelic-logging.lowDataMode" .)`, which is not an "empty string" so it is `true` if it is used
+        as an evaluation somewhere else.
+    */}}
+    {{- .Values.lowDataMode -}}
+  {{- end -}}
+{{- else -}}
+{{/* This allows us to use `$global` as an empty dict directly in case `Values.global` does not exists */}}
+{{- $global := index .Values "global" | default dict -}}
+{{- if get $global "lowDataMode" | kindIs "bool" -}}
+  {{- if $global.lowDataMode -}}
+    {{- $global.lowDataMode -}}
+  {{- end -}}
+{{- end -}}
```

Comment:
We could also just `if .Values.global`, right?
```suggestion
{{- if .Values.global -}}
  {{- if get $global "lowDataMode" | kindIs "bool" -}}
    {{- if $global.lowDataMode -}}
      {{- $global.lowDataMode -}}
    {{- end -}}
  {{- end -}}
{{- end -}}
```

Reply:
I thought that with a variable it was clearer. Fixed.

Reply:
I think in "real code" is clearer, but in templates... Variables in templates feel wrong to me.

Just a personal thought tho. Thanks!

Reply:
EDIT: My suggestion is actually wrong, it would not allow for the `else` to evaluate as it should. Please ignore it.